src/surface.cpp

```diff
@@ -279,7 +277,13 @@ Direction DAGSurface::reflect(Position r, Direction u, Particle* p) const
 {
   Expects(p);
   p->history_.reset_to_last_intersection();
-  p->last_dir_ = Surface::reflect(r, u, p);
+  moab::ErrorCode rval;
+  moab::EntityHandle surf = dagmc_ptr_->entity_by_index(2, dag_index_);
+  double pnt[3] = {r.x, r.y, r.z};
+  double dir[3];
+  rval = dagmc_ptr_->get_angle(surf, pnt, dir, &p->history_);
```

Comment:
why not use direction?

Reply:
looks like its uninitialised here?

Reply:
nope, ignore me - dir is being set in the get_angle funciton - sorry :)

Reply:
Yikes. Yup. Thanks!